### ISSUE TITLE
Error - Could not run aider on existing git repo

### ISSUE THREAD

**Issue by distributev** [paul-gauthier/aider#501]
Using the latest version of aider and the script

```
cd /correct/path/to/existing/git/repo
export OPENAI_API_KEY=sk_correct_api_key
aider --model gpt-4-0125-preview
```

I get the following error. I can confirm that the network is fine and that there are credits for the API_KEY.

Does aider tries to send any information about the git repo first time that is run - Could this error happen because the git repo is large?

![image](https://github.com/paul-gauthier/aider/assets/19224635/55be64dc-9b1e-499b-9f45-441c881bf3f9)

**Comment by paul-gauthier** [comment]
Thanks for trying aider and filing this issue.

It looks like there is a problem connecting to a server involved with the tiktoken library. Either your local network environment is preventing the connection or the server was down.

**Comment by paul-gauthier** [comment]
That error message appears to have nothing to do with the repo size. It an an HTTP connection error. The attempt to make an HTTP connection is not succeeding within the allotted time.

**Comment by distributev** [comment]
OK, thank you I'll try tomorrow and double check that there are no network related obstacles. Thank you for aider, I watched couple of videos on it and I'm eager to try it.

PS - recently I worked with another python/openai related library which was also displaying similar network errors http 404 but actually the root cause was totally unrelated just that the root error was wrapped/displayed as network related error. I saw the root error only after I used curl to call the same end point API.

**Comment by distributev** [comment]
Hello again,

This morning it is still failing with the same error message as the days before.

From the same command line interface from where aider is not working I did a successful curl request to the https://api.openai.com/v1/chat/completions api so I don't think there is a network error.

In the context of aider what is the role of the tiktoken library executed at the time of 
```aider --model gpt-4-0125-preview``` ?

do you know what would be the equivalent request url command which aider/tiktoken is executing and which I can use curl to test in the same way I tested https://api.openai.com/v1/chat/completions ?

![image](https://github.com/paul-gauthier/aider/assets/19224635/0d6c1110-f6bb-4256-90e7-ba578ec3c030)

**Comment by distributev** [comment]
I did couple of more investigations and found this

Could that be the reason? Did you see this before?

![image](https://github.com/paul-gauthier/aider/assets/19224635/1caeba2a-126f-4c77-bb95-d86975c8d04a)

**Comment by paul-gauthier** [comment]
I'm going to close this issue for now, but feel free to add a comment here and I will re-open or file a new issue any time.
